apps/query-ui/queryui/queryui.py

```diff
@@ -46,34 +46,40 @@ def show_schema(container: Any, schema: Dict[str, Tuple[str, Set[str]]]):
         st.dataframe(table_data)
 
 
-def show_traces():
-    """Show the traces in the given trace_dir."""
-    st.session_state.query_trace_dir = f"{st.session_state.trace_dir}/{st.session_state.query_id}"
-    for id in sorted(os.listdir(f"{st.session_state.query_trace_dir}")):
-
-        # Initialize a list to hold the data
+def show_traces(trace_dir):
+    """Show the traces in the trace_dir."""
+    for node_id in sorted(os.listdir(trace_dir)):
         data_list = []
-        directory = f"{st.session_state.query_trace_dir}/{id}"
-
+        directory = os.path.join(trace_dir, node_id)
         for filename in os.listdir(directory):
             f = os.path.join(directory, filename)
             if os.path.isfile(f):
                 with open(f, "rb") as file:
                     doc = pickle.load(file)
-                    doc_list = doc.properties["entity"]
-
-                    for p in doc.properties:
-                        if p not in BASE_PROPS and p in doc.properties:
-                            doc_list[p] = doc.properties[p]
-                    doc_list["text_representation"] = doc.text_representation
-                    data_list.append(doc_list)
+                    # For now, skip over MetadataDocuments.
+                    if "doc_id" not in doc:
+                        continue
+                    data_list.append(doc)
 
         df = pd.DataFrame(data_list)
```

Comment:
is it possible to let each doc.properties and doc.properties.entity have its own column? helps when debugging so you can tell if something is wrong from a quick glance

Reply:
Yes, we can tweak it, but I am working on a new UI for this anyway.